internal/services/storage/storage_account_resource.go

```diff
@@ -1728,22 +1720,17 @@ func resourceStorageAccountRead(d *pluginsdk.ResourceData, meta interface{}) err
 		d.Set("enable_https_traffic_only", props.EnableHTTPSTrafficOnly)
 		d.Set("is_hns_enabled", props.IsHnsEnabled)
 		d.Set("nfsv3_enabled", props.EnableNfsV3)
+		d.Set("cross_tenant_replication", props.AllowCrossTenantReplication)
```

Comment:
is this returned from the API for older storage accounts/where this isn't set? my guess is it'd be nil, which would get translated to `false` here (leading to a diff)